docs: update PHPUnit testing README.md with corrections #62065 (#7389)
WordPress/wordpress-develop: tests/phpunit/README.md

# PHPUnit Tests

PHPUnit is the official testing framework chosen by the core team to test our PHP code.

This documentation will assume you have a local development environment of your own choosing. For other testing workflows, such as Docker, see the [PHP: PHP Unit](https://make.wordpress.org/core/handbook/testing/automated-testing/phpunit/) handbook page.

## Setup

> 1. Check out the wordpress-develop repository
> 
>        git clone git://develop.git.wordpress.org/ wordpress-develop

Reviewer: ```suggestion
1. Check out the [`wordpress-develop` repository](https://github.com/WordPress/wordpress-develop):

       git clone git@github.com:WordPress/wordpress-develop.git
```
I wonder if this should instead refer to the Github repo, as PRs there are the typical way to contribute patches and tests.

>        cd wordpress-develop

Reviewer: ```suggestion
cd wordpress-develop
```
For this and the `git clone` line above it, please wrap with three backticks instead of spaces ([for example](https://github.com/WordPress/wordpress-develop/blob/552077a27589294b2cf6097205a4045585016281/README.md?plain=1#L50-L55)).

> 2. Run `composer update --with-all-dependencies`

Reviewer: ```suggestion
2. Run `composer update --with-all-dependencies`.
```
Nit.

3. Create a clean MySQL database and user. **DO NOT USE AN EXISTING DATABASE** or you will lose data, guaranteed.
4. Copy `wp-tests-config-sample.php` to `wp-tests-config.php`, edit it to include your testing database credentials.
5. Run the tests from the repository root:

>    - To execute a particular test:
> 
>          $ vendor/bin/phpunit tests/phpunit/tests/test_case.php
>    - To execute all tests:
> 
>          $ vendor/bin/phpunit

Reviewer: ```suggestion
- To execute a particular test:

vendor/bin/phpunit tests/phpunit/tests/test_case.php

- To execute all tests:

vendor/bin/phpunit

```
Suggest dropping the terminal prompt character. For these commands, please substitute the block formatting spaces by wrapping with three backticks for [consistency with other docs](https://github.com/WordPress/wordpress-develop/blob/552077a27589294b2cf6097205a4045585016281/README.md?plain=1#L50-L55) (I'm unsure how to add these within a backticked suggestion block 😅).

Author: I have switched to the three backticks, however this still requires indentation or else the list formatting gets messed up. I hope that's okay! :)

# PHPUnit Tests

PHPUnit is the official testing framework chosen by the core team to test our PHP code.

This documentation will assume you have a local development environment of your own choosing. For other testing workflows, such as Docker, see the [PHP: PHPUnit](https://make.wordpress.org/core/handbook/testing/automated-testing/phpunit/) handbook page.

## Setup

1. Check out the [`wordpress-develop` repository](https://github.com/WordPress/wordpress-develop):
	```
	git clone git@github.com:WordPress/wordpress-develop.git
	cd wordpress-develop
	```

> 2. Run `composer update --with-all-dependencies`.

Reviewer 2: ```suggestion
2. Run `composer update -W`.
```

This is the shorthand for this option.

Author: Is that preferred here? I personally default to [avoiding short options](https://architecture.lullabot.com/adr/20211006-avoid-command-aliases/), especially in documentation. Happy to make this change if this is not the  case for WordPress documentation. ☺️

> 3. Create a clean MySQL database and user. **DO NOT USE AN EXISTING DATABASE** or you will lose data, guaranteed.

Reviewer 2: ```suggestion
3. Create a clean MySQL database and user. **DO NOT USE AN EXISTING DATABASE.** The test database is emptied and repopulated every time tests are run.
```

> 4. Copy `wp-tests-config-sample.php` to `wp-tests-config.php`, edit it to include your testing database credentials.

Reviewer 2: ```suggestion
4. Copy `wp-tests-config-sample.php` to `wp-tests-config.php` and enter your database credentials. **Again, use a separate database.**
```
